packages/adapters/wagmi/src/client.ts

```diff
@@ -428,6 +432,32 @@ export class WagmiAdapter implements ChainAdapter {
           return BigInt(0)
         }
       },
+
+      grantPermissions: async params => {
+        if (!this.wagmiConfig) {
+          throw new Error('connectionControllerClient:request - wagmiConfig is undefined')
+        }
+
+        const connections = getConnections(this.wagmiConfig)
+        console.log('>> Wagmi grantPermissions - connections', connections)
+
+        const connection = connections[0]
+
+        if (!connection?.connector) {
+          throw new Error('connectionControllerClient:request - connector is undefined')
+        }
+
+        const provider = (await connection.connector.getProvider()) as Awaited<
+          ReturnType<(typeof UniversalProvider)['init']>
+        >
```

Comment:
Is this not the same?
```suggestion
        const provider = (await connection.connector.getProvider()) as UniversalProvider
```